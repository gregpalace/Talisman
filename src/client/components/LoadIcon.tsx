import React from 'react';

import '../../stylesheets/loading.scss';

const LoadIcon = () => {
  return <div className='loading' />;
};

export default LoadIcon;
