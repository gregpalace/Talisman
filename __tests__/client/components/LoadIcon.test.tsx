import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import LoadIcon from '../../../src/client/components/LoadIcon';

describe('React unit tests', () => {
  let wrapper: any;

  beforeAll(() => {
    wrapper = shallow(<LoadIcon />);
  });

  describe('<LoadIcon />', () => {
    it('should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a div with className of "loading"', () => {
      expect(wrapper.find('div')).toBeDefined();
      expect(wrapper.find('div').prop('className')).toEqual('loading');
    });
  });
});
