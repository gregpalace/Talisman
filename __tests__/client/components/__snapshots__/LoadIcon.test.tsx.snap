// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests <LoadIcon /> should render correctly 1`] = `
<div
  className="loading"
/>
`;
